### PR TITLE
Hive TRV Support

### PR DESCRIPTION
**Pull request by MagicalTrev89**
The changes I have proposed enable TRV support. This enables:
Discovery of TRV devices
Current Temperature
Target Temperature
Adjusting temperature
Changing mode, on, off, schedule

This PR does not cater for the boost option which will work on separately if this is approved.

### CODE REVIEW

**Comment by Rendili** [comment]
Hi @MagicalTrev89 ,

Sorry for the delayed reply.
There has been a big change regarding the climate platform in home-assistant which we have been trying to adapt to.
Soon after we have managed to go live with the climate / water_heater changes we can look review and merge the TRV support.

Do you also use Home Assistant and need for these to be added there too?
What sort of functions values does this support?

Thanks,

**Comment by MagicalTrev89** [comment]
@Rendili  No problem! 

Yeah I also have the required changes for Homeassistant, on my setup i have cloned the hive component and the pyhiveapi and imported as a custom_component with the correct changes

I've had it running for several weeks now and the TRVs are working fine. Only bit i havent done is the heating boost on the TRVs

You get:
Current Temperature
Target Temperature
Mode: Auto, On, Off (Boost)

Basically the same attributes that you get from the main Hive thermostat

**Comment by MagicalTrev89** [comment]
@Rendili I have made further updates to this to support the changes you have made for HA 0.96, 
 I have just committed my changes, please take a look, i have also made the relevant updates to the hive component and have these ready to go.

I have gone further on this and added support for HVAC action so you can now see if the heating is idle or actually heating.

**Comment by Rendili** [comment]
@MagicalTrev89 
Please could you create a new pull request against the `TRV-Support` branch, I will merge straight away to there and we can test on this.
Also, would you be able to make your HA component update changes to a `TRV-Support` branch on my fork of home-assistant? If so I will create this branch (I need to rebase first) and we can do a full end to end test

**Comment by MagicalTrev89** [comment]
I've done the pyhiveapi as requested and i see you have merged.

I cant push any changes to your homeassistant TRV-Support branch as access denied

**Comment by Rendili** [comment]
Are you able to do a pull request to that branch?

**Comment by MagicalTrev89** [comment]
Alright, done, you should see it now on your TRV-Support branch for HA.

**Comment by Rendili** [comment]
great, thanks
although I see two conflicts so it cant merge at the moment, checking now

**Comment by Rendili** [comment]
going to close this PR has it has been superseded
